### ISSUE TITLE
Pekko requires non-existent dependency version

### ISSUE THREAD

**Issue by jonathan-neufeld-asurion** [apache/incubator-pekko-http#327]
Unable to spin up an instance of Pekko HTTP Server because of an apparent version conflict, effectively requiring version 1.0.1, which is not available:
```
Starting server failed because of Current version of Pekko is [1.0.0], but pekko-http requires version [1.0.1]
```
While there is probably some separate issue in the dependency graph giving rise to this, what is a priori is that a Pekko release should not require a version that does not yet exist or is not yet published.

### Dependency Graph
It's not clear to me from the dependency graph what would cause this.  This graph is taken from [this branch](https://github.com/softwaremill/tapir/pull/3226).
![image](https://github.com/apache/incubator-pekko-http/assets/73544551/55b74bac-3963-40ca-85f0-079aea16ca76)

**Comment by mdedetrich** [comment]
This is likely due to https://github.com/apache/incubator-pekko-http/blob/e7da6d3876300efbe9a2e6c6105cdc230f55d3b1/project/PekkoDependency.scala#L54, bumping this version seems to have been an oversight.

@pjfanning @jrudolph I think that downgrading this to 1.0.0 should fix the issue?

**Comment by pjfanning** [comment]
> This is likely due to
> 
> https://github.com/apache/incubator-pekko-http/blob/e7da6d3876300efbe9a2e6c6105cdc230f55d3b1/project/PekkoDependency.scala#L54
> 
> , bumping this version seems to have been an oversight.
> @pjfanning @jrudolph I think that downgrading this to 1.0.0 should fix the issue?

I think we need to find a way so that when we run the pekko-http unit tests that it uses the latest pekko core 1.0.x jars but we should fix the version check done at runtime so that any pekko core 1.0.x jars is accepted.

**Comment by mdedetrich** [comment]
> I think we need to find a way so that when we run the pekko-http unit tests that it uses the latest pekko core 1.0.x jars but we should fix the version check done at runtime so that any pekko core 1.0.x jars is accepted.

This should already be happening due to the code in https://github.com/apache/incubator-pekko-http/blob/e7da6d3876300efbe9a2e6c6105cdc230f55d3b1/project/PekkoDependency.scala#L96-L123 which by default builds Pekko against the latest snapshot.

FYI, this error is coming from https://github.com/apache/incubator-pekko-http/blob/43b9ca9c2eb989e2acca926a2fdc6a42dc82053d/http-core/src/main/scala/org/apache/pekko/http/scaladsl/Http.scala#L73 which is derived from https://github.com/apache/incubator-pekko-http/blob/e7da6d3876300efbe9a2e6c6105cdc230f55d3b1/project/PekkoDependency.scala#L54

**Comment by pjfanning** [comment]
@jonathan-neufeld-asurion v1.0.1 of pekko core jars is available. They were released before pekko-http 1.0.0 was released.

Have a look https://mvnrepository.com/artifact/org.apache.pekko

You'll see 1.01 jars for `pekko-actor-*`, `pekko-stream-*`, `pekko-remote`, etc. All the libs that are built for this repo:

https://github.com/apache/incubator-pekko

**Comment by pjfanning** [comment]
The nightly tests ran with the wrong Pekko Core jars last night.

https://github.com/apache/incubator-pekko-http/actions/runs/6477130954/job/17586840327

[info] Building Pekko HTTP 1.1.0-M0+11-9de3848d+20231011-0215-SNAPSHOT against Pekko 1.0.0 on Scala 2.12.18

**Comment by pjfanning** [comment]
I think for 1.0.x that we should test with the latest 1.0.x releases or snapshots and not with Pekko 1.0.0. I can agree that the runtime check for Pekko version can be relaxed to accept Pekko 1.0.0 but we should be testing with Pekko 1.0.1 or a 1.0.1+123 snapshot.

**Comment by mdedetrich** [comment]
Let's make a seperate issue to discuss this then. What is done now is the same as how it's historically been in Akka, i.e. main build is built against the minimum supported version and nightly against snapshots.

**Comment by jonathan-neufeld-asurion** [comment]
> @jonathan-neufeld-asurion v1.0.1 of pekko core jars is available. They were released before pekko-http 1.0.0 was released.
> 
> Have a look https://mvnrepository.com/artifact/org.apache.pekko
> 
> You'll see 1.0.1 jars for `pekko-actor-*`, `pekko-stream-*`, `pekko-remote`, etc. All the libs that are built for this repo:
> 
> https://github.com/apache/incubator-pekko

Thank you @pjfanning, I see **only 1.0.0** of `pekko-http` available, is this correct?
![image](https://github.com/apache/incubator-pekko-http/assets/73544551/b1aac654-4485-45d0-937b-44f8537eb131)

**Comment by pjfanning** [comment]
@jonathan-neufeld-asurion pekko-http has its own release schedule - v1.0.0 is the only release of pekko-http. I'm not sure if you ever used akka-http but this is also the case with akka-http.
